app/usecases/performance.py

```diff
@@ -20,25 +22,31 @@ class PerformanceScore(TypedDict):
     miss_count: int
 
 
+@retry(
+    retry=retry_if_exception_network_related(),
+    wait=wait_exponential(),
+    stop=stop_after_attempt(10),
+    reraise=True,
+)
 async def calculate_performances(
     scores: list[PerformanceScore],
 ) -> list[tuple[float, float]]:
     response = await app.state.services.http_client.post(
         f"{config.PERFORMANCE_SERVICE_URL}/api/v1/calculate",
         json=scores,
     )
-    if not response.is_success:
-        logging.error(
-            "Performance service returned non-2xx code on calculate_performances",
-            extra={"status": response.status_code},
-        )
-        return [(0.0, 0.0)] * len(scores)
+    response.raise_for_status()
```

Comment:
i think i agree with moving to response.raise_for_status() to hard fail requests rather than setting 0